raco/algebra.py

```diff
@@ -366,7 +381,29 @@ def shortStr(self):
 """Logical Relational Algebra"""
 
 
-class Union(BinaryOperator):
+class IdenticalSchemeBinaryOperator(BinaryOperator):
+
+    """BinaryOperator where both sides have the same schema"""
+
+    def partitioning(self):
+        """attributes that are partitioned in both inputs"""
+        return RepresentationProperties(hash_partitioned=set.intersection(
```

Comment:
Agreed, I didn't give this much thought; maybe I'll add an example to the comments